services/authoring/src/repositories/apps-repository.ts

```diff
@@ -33,12 +40,89 @@ export default class AppsRepository {
     return this.apps;
   }
 
-  async saveApp(appId, manifest, author) {
-    return await this._gitTransactionManager.write(async repo => {
+  async createApp(appId: string, manifest: AppManifest, author) {
+    return await this._gitTransactionManager.write(async (repo) => {
       await repo.updateFile(`external_apps/${appId}.json`, JSON.stringify(manifest, null, 4));
       const commitId = await repo.commitAndPush(`created app ${appId}`, author);
       await this.refresh();
       return commitId;
     });
   }
+
+  async updateApp(appId: string, manifest: Partial<Pick<AppManifest, 'name' | 'permissions'>>, author) {
```

Comment:
Where does updateApp is being called?

Reply:
Oops.. missed it.. I'll add endpoint

Reply:
Not sure we need it, "permissions" updating is legacy (all is through the gateway) and app name can be immutable. (From permission perspective, it removes the risk of changing the name for impersonation)